vignettes/admiral.Rmd

```diff
@@ -19,135 +19,69 @@ library(admiraldev)
-First, we will load our packages and example datasets to help with our `ADSL` creation. The `{dplyr}` and `{lubridate}` packages are `{tidyverse}` packages and used heavily throughout this script.  The `{admiral}` package also leverages the `{pharmaversesdtm}` package for example SDTM datasets which are from the CDISC Pilot Study.
+In this vignette we will explore some of the different type of derivation functions 
+offered by `{admiral}`, as well as the argument conventions they follow and how best 
+to start an `{admiral}` script. To help with the examples showcased, we will load 
+some packages and set up some example data to manipulate later. The `{dplyr}` and 
+`{lubridate}` and `{stringr}` packages are `{tidyverse}` packages and are used heavily
+throughout this vignette.  The `{admiral}` package also leverages the `{pharmaversesdtm}` 
+package for example SDTM datasets which are from the CDISC Pilot Study.
 
 ```{r, message=FALSE, warning=FALSE}
 library(dplyr, warn.conflicts = FALSE)
 library(lubridate)
-library(admiral)
+library(stringr)
+library(tibble)
 library(pharmaversesdtm)
+library(admiral)
 
 # Read in SDTM datasets
 data("dm")
 data("ds")
 data("ex")
+data("vs")
+data("admiral_adsl")
 
 dm <- convert_blanks_to_na(dm)
 ds <- convert_blanks_to_na(ds)
 ex <- convert_blanks_to_na(ex)
-```
+vs <- convert_blanks_to_na(vs)
 
-## Derive Treatment Variables (`TRT0xP`, `TRT0xA`) {#treatmentvargs}
-
-The mapping of the treatment variables is left to the ADaM programmer. An example mapping may be:
-
-```{r eval=TRUE}
-adsl <- dm %>%
-  mutate(TRT01P = ARM, TRT01A = ACTARM)
-```
-
-## Derive/Impute Numeric Treatment Date/Time and Flag Variables (`TRTSDTM`, `TRTEDTM`, `TRTSTMF`, `TRTETMF`) {#trtdatetimegs}
-
-The function `derive_vars_dtm()` can be used to convert the DTC variables from
-`EX` to numeric datetime variable and impute missing components. The function
-call returns the original data frame with the column `EXSTDTM`, `EXENDTM` and
-corresponding time imputation  flag variables `EXSTTMF` and  `EXENTMF` added to
-the end of the dataframe. Exposure observations with an incomplete date are
-ignored. We impute missing time to be 23:59:59 using `time_imputation = "last"`.
-The required imputation flags are determined automatically by the function. Here
-only time imputation flags are derived because time is imputed but date is not
-imputed.
-
-Don't be intimidated by the number of arguments! We try to make our arguments
-self-explanatory, e.g. the `new_vars_prefix` places `EXST` at the start of the
-`--DTM` variable and `time_imputation = "last"` appends 23:59:59.  However, this
-is not always possible to make every argument self-explanatory.  If you click on
-the function, `derive_vars_dtm()`, you can bring up the reference documentation
-and learn more about each argument.
-
-```{r, eval=TRUE}
-# Derive treatment variables
-## Impute time of exposure dates (creates numeric datetime and time imputation flag variables)
 ex_ext <- ex %>%
```

Comment:
What about breaking this up a bit?  Load and say why we convert blanks to NA - a huge stuggle for new users

then move into the derive functions
![image](https://github.com/pharmaverse/admiral/assets/10111024/d34f8622-4938-4f9f-9246-2ddbda28c667)

Reply:
what about a little note about us preparing datasets to showcase adding variables for ADSL and adding records for ADVS?

Reply:
I would remove the `convert_blanks_to_na()` chunk. It is not necessary for pharmaversesdtm datasets and only necessary for hybrid workflows where the SDTMs are created with SAS and the ADaMs with R. Thus we don't need to present it on the "Get Started" page.

Reply:
happy to remove for simplicity, vignette is already getting quite long